### DISCUSSION TITLE
How to run a built onnx neural network on CPU?

### DISCUSSION THREAD

**Discussion by FlintWangacc** [iree-org/iree#17813] [General]
I use ```iree-import-onnx`` to transform an onnx file into a mlir file
```shell
iree-import-onnx $PATH/models/Computer_Vision/botnet26t_256_Opset16_timm/regnetv_064_Opset16.onnx -o ./regnetv_064_Opset16.mlir
```
Then I  use ```iree-compile``` to compile this mlir file
```shell
../iree-build/install/bin/iree-compile --iree-hal-target-backends=vmvx ./regnetv_064_Opset16.mlir -o ./regnetv_064_Opset16.vmfb
```
These operations all went well.

But when I try to run this neural network, I don't know what is the entry function. I didn't find any documentation about this.

```shell
../iree-build/tools/iree-run-module   --device=local-task   --module=./regnetv_064_Opset16.vmfb   --function=$ENTRY_FUNCTION --input="1x224x224x3xf32=0"
```

Does this compile process need any new argument or I just need to find the **$ENTRY_FUNCTION**

**Comment by ScottTodd** [comment]
You need to specify the entry function, yes. The function name is whatever you construct it as in the original program, or what the existing model used. You can find that by
* reading documentation for the program
* inspecting the .onnx file (netron, Python API, etc.)
* inspecting the .mlir file (if text, not binary, look for a `func.func public @ENTRY_FUNCTION_NAME_HERE`)
* inspecting the .vmfb file (e.g. using `iree-dump-module`)
